### DISCUSSION TITLE
Can/Should I predefine actors at startup

### DISCUSSION THREAD

**Discussion by jababda** [akkadotnet/akka.net#6537] [Q&A]
I have potentially an odd question. 
I need to build a little simulation of a material coming into a pipeline, moving though various processes with branching effects and finally being deposited somewhere. 

A colleague recommended akka as we can easily represent the tree structure of the connected pipes i.e.
ingestion -> pipe -> material splitter -> material 1 goes to store 1
                                                            -> material 2 goes to the bin

Because the actors are known before the application starts I would want to setup the tree of actors before I start sending events. 
This leads to some interesting questions:
1. Should I be using Akka for this and does it make sense to define the tree upfront.
2. Does each actor have access to its children and can they deduce type. (needed for the material splitter)
3. Are there any other options or alternatives the community would recommend?

**Comment by malclear** [comment]
Top-level actors can be defined at startup. To have parent-child relationships between actors, you'll need to declare the children from the parent, of course. This can be done as needed while the application runs, including immediately after the parent actor is created.

**Reply by jababda** [reply]
Is there any issues with top level actors communicating? On further evaluation I found my tree structure actually started interconnecting.
